src/ex4/server/routes/api.js

```diff
@@ -1 +1,48 @@
-// Define your endpoints here (this is your "controller file")
\ No newline at end of file
+// Define your endpoints here (this is your "controller file")
+import { Router } from "express";
+import express from "express";
+import ItemManager from "../services/item_manager.js";
+const router = express.Router();
+const item_manager = new ItemManager();
+
+router.route("/").get(async (req, res) => {
+	try {
+		const allItems = await item_manager.getAllItems();
+		if (!allItems.data) {
+			throw new Error("something went wrong.");
+		}
+		res.status(200).send(allItems.data);
+	} catch (err) {
+		res.status(500).send(err.message);
+	}
+});
+
+router.route("/add").post(async (req, res) => {
+	try {
+		const allItems = await item_manager.validateInput(req.body);
+		if (!allItems) {
+			throw new Error("something went wrong.");
+		}
+		res.status(201).json(allItems).send(allItems.data);
+	} catch (err) {
+		console.log(err);
+		res.status(406).send(err.message);
```

Comment:
why 406?

Reply:
I read about it and decided to change it to 400